cmdstanpy/model.py

```diff
@@ -276,33 +226,25 @@ def stan_file(self) -> OptionalPath:
         return self._stan_file
 
     @property
-    def exe_file(self) -> OptionalPath:
+    def exe_file(self) -> Union[os.PathLike, str]:
```

Comment:
I believe the only places that `self._exe_file` is set now are lines 135 and 171. Since both `os.path.realpath` and `compilation.compile_stan_file` return `str`, this can probably be set to simply `def exe_file(self) -> str`.